packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -356,10 +348,32 @@ export class CfnInclude extends core.CfnElement {
     const options: core.FromCloudFormationOptions = {
       finder,
     };
+    const l1ClassFqn = cfn_type_to_l1_mapping.lookup(resourceAttributes.Type);
+    let l1Instance;
+
+    if (l1ClassFqn) {
+      const [moduleName, ...className] = l1ClassFqn.split('.');
+      const module = require(moduleName);  // eslint-disable-line @typescript-eslint/no-require-imports
+      const jsClassFromModule = module[className.join('.')];
+
+      l1Instance = this.nestedStacksToInclude[logicalId]
```

Comment:
I found a bug!

The following test:

```ts
  test('throws an exception when and ID was passed in nestedStacks that is a resource type not in the CloudFormation schema', () => {
    expect(() => {
      new inc.CfnInclude(stack, 'Template', {
        templateFile: testTemplateFilePath('custom-resource.json'),
        nestedStacks: {
          'CustomResource': {
            templateFile: testTemplateFilePath('whatever.json'),
          },
        },
      });
    }).toThrow(/Nested Stack with logical ID 'CustomResource' is not an AWS::CloudFormation::Stack resource/);
  });
```

where `custom-resource.json` is:

```json
{
  "Resources": {
    "CustomResource": {
      "Type": "AWS::CustomResource::Type"
    }
  }
}
```

fails with: `Received function did not throw`.

I'll leave it to you to figure out what happened, and fix it 🙂.